### PR TITLE
[RHCLOUD-19160] Vault mock update

### PR DESCRIPTION
**Pull request by petracihalova**
we need to update vault mock methods List() and Read() because we need it for writing tests for these functions (when secret store = vault):
- EndpointListAuthentications() from endpoint handler ([RHCLOUD-18283](https://issues.redhat.com/browse/RHCLOUD-18283)) 
- ApplicationListAuthentications() from app handler ([RHCLOUD-18281](https://issues.redhat.com/browse/RHCLOUD-18281))
- SourceListAuthentications() from source handler (done)
- ApplicationAuthenticationListAuthentications() from app auth handler ([RHCLOUD-18282](https://issues.redhat.com/browse/RHCLOUD-18282))

so basically we need more test data for these tests and Vault mock that is able to handle them

and

after changes in first commit some tests failed, so second commit fixes this

--------------

This change was originally part of https://github.com/RedHatInsights/sources-api-go/pull/278

**LINKS:** [RHCLOUD-19160](https://issues.redhat.com/browse/RHCLOUD-19160)

### CODE REVIEW

**Review by MikelAlejoBR — APPROVED** [review]

**Review by lindgrenj6 — APPROVED** [review]

**Comment by petracihalova** [comment]
rebased

**Review by lpichler — APPROVED** [review]
